core/src/main/java/io/questdb/std/Chars.java

```diff
@@ -482,14 +482,32 @@ public static String toLowerCaseAscii(@Nullable CharSequence value) {
             b.put(toLowerCaseAscii(value.charAt(i)));
         }
         return b.toString();
-
-
     }
 
     public static char toLowerCaseAscii(char character) {
         return character > 64 && character < 91 ? (char) (character + 32) : character;
     }
 
+    public static String toUpperCaseAscii(@Nullable CharSequence value) {
+        if (value == null) {
+            return null;
+        }
+        final int len = value.length();
+        if (len == 0) {
+            return "";
+        }
+
+        final CharSink b = Misc.getThreadLocalBuilder();
+        for (int i = 0; i < len; i++) {
+            b.put(toUpperCaseAscii(value.charAt(i)));
+        }
+        return b.toString();
```

Comment:
This implementation is needlessly allocating. Calls to this method are made from function, that has local CharSink already. String instantiation is not required.